### PR TITLE
Separate package `pkg/apis` and `pkg/client` into submodules

### LINKED ISSUE #2531: publish API and generated client as a separate package
## Feature Request

**Is your feature request related to a problem? Please describe:**

This is useful for external projects to vendor our API and client.

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

We can move them into a go submodule (see [example](https://github.com/go-modules-by-example/submodules)) and release it with the root module (tidb-operator).

### LINKED ISSUE #3804: separate the tidb-operator api into a dedicated go module
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, if my app want to import the api of tidb-operator (`pkg/apis/pingcap/v1alpha1`), I have to import the whole tidb-operator module. However, importing `tidb-operator` is hard:

```
go get github.com/pingcap/tidb-operator@v1.1.9
go get: github.com/pingcap/tidb-operator@v1.1.9 requires
	github.com/uber-go/atomic@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

I have to copy all the `replace` directives in the `go.mod` file of my app in order to import `tidb-operator`, what is even worser is the `tidb-operator` module will introduce lots of in-direct dependencies to my module, which makes the dependency tree a mass.

**Describe the feature you'd like:**

Like Kubernetes (`k8s.io/api`), separate the api to a dedicated module, for example, `pingcap/tidb-operator/api`, the dependencies of this module can be rather simple since it only contains the type definition.

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #2531 
Close #3804

Separate package `pkg/apis` and `pkg/client` into submodules.

Ref #4129 #4126 #4125 #4119 #4118

### What is changed and how does it work?

* Remove dependency about package `advanced-statefulset` in `pkg/apis`
* Separate package `pkg/apis` into submodule `github.com/pingcap/tidb-operator/pkg/apis`.
* Separate package `pkg/client` into submodule `github.com/pingcap/tidb-operator/pkg/client`.
* Modify Makefile to support multi modules.
* Add `hack/verify-modules.sh` to check if all packages in submodules are same as main module.

### Code changes

- [ ] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Separate package `pkg/apis` and `pkg/client` into submodules
```

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by handlerww — APPROVED** [review]

File: pkg/client/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/pingcap/tidb-operator/pkg/client
+
+go 1.13
+
+require (
+	github.com/pingcap/tidb-operator/pkg/apis v0.0.0
+	k8s.io/apimachinery v0.0.0-20190913080033-27d36303b655
```
Comment (liubog2008) [inline pkg/client/go.mod:7]: Change to use a stable version?

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**